### ISSUE TITLE
Add timestamp to saved support bundle files

### ISSUE THREAD

**Issue by markpundsack** [replicatedhq/troubleshoot#228]
Inspired by https://gitlab.com/gitlab-com/support/toolbox/kubesos which adds a timestamp to the saved filename. We at least do support multiple file generation by incrementing the filename like `support-bundle (1).tar.gz`, but `support-bundle-<timestamp>.tar.gz` may be more robust.

**Comment by divolgin** [comment]
@markpundsack is timestamp a unix timestamp? or if it's human-readable, what should be format/precision?  Should we go with RFC 3339 format?

**Comment by markpundsack** [comment]
Good question. KubeSOS uses:
```
timestamp="$(date +%d%m%Y%H%M%S)"
```
which is not particularly human readable nor an industry standard, but does result in relatively unique files sorted correctly.

I'm a fan of ISO dates, but ISO times get more complicated. UTC is not human-friendly. ISO allows local timezone to be the default. I guess we need second-level precision, but I'd want to avoid sub-second precision.

So, "2007-04-05T14:30".

**Comment by matias-manavella-nimble** [comment]
@divolgin @markpundsack Hi! I've modified the code to obtain a timestampt like the one Mark described in the bundle file:

`$ ./troubleshoot https://support-bundle.replicated.com`
`support-bundle-2020-08-03T16:52.tar.gz`

I will make a PR so you can check it! Time zone and further precision may be added as well!

**Comment by markpundsack** [comment]
@manavellamnimble Thanks! Does it still handle the unlikely collision of a file existing with the same name? Current version appends "(2)", which while not ISO, would be sufficient for our purposes, I think.

**Comment by matias-manavella-nimble** [comment]
@markpundsack I added a second-level precision timestamp, but in the unlikely case of a collision, a number will be added! I tested it with the previous timestamp:

```
-rw-r--r--  1 admin  staff   241K Aug  3 17:43 support-bundle-2020-08-03T17:43 (1).tar.gz
-rw-r--r--  1 admin  staff   241K Aug  3 17:43 support-bundle-2020-08-03T17:43 (2).tar.gz
-rw-r--r--  1 admin  staff   241K Aug  3 17:43 support-bundle-2020-08-03T17:43 (3).tar.gz
-rw-r--r--  1 admin  staff   241K Aug  3 17:43 support-bundle-2020-08-03T17:43 (4).tar.gz
-rw-r--r--  1 admin  staff   241K Aug  3 17:43 support-bundle-2020-08-03T17:43.tar.gz
```

New format would look like this:

```
$ ./troubleshoot https://support-bundle.replicated.com       
support-bundle-2020-08-03T17:57:25.tar.gz
```

**Comment by manavellamnimble** [comment]
New format would look like this:

```
$ ./troubleshoot https://support-bundle.replicated.com       
support-bundle-2020-08-03T17:57:25.tar.gz
```

**Comment by markpundsack** [comment]
@manavellamnimble Perfect, thanks!

**Comment by matias-manavella-nimble** [comment]
Hi there! I think this issue may be closed now, as it was solved in the PR #230 merged yesterday
